### DISCUSSION TITLE
"newStreamAvailable" for same streamId is getting called multiple times.

### DISCUSSION THREAD

**Discussion by MBFLYT** [ant-media/Ant-Media-Server#5858] [Q&A] [answered]
"newStreamAvailable" for same streamId is getting called multiple times, and sometimes the audio is not getting played in some device, I am using Antmedia SDK for JS in my Angular and I am following this sample code https://github1s.com/ant-media/conference-call-application/blob/97c496ea053afd171363f051de8a573ec78ee37e/react/src/pages/AntMedia.js#L360

**Comment by USAMAWIZARD** [comment]
Hi @MBFLYT ,
is there any reproduce scenario  for the issue.

**Comment by MBFLYT** [comment]
Hey @USAMAWIZARD 

This happens when I try to play the stream using the streamId, but this stopped when I started use multitrack.

So is multitrack is the recommended solution for conference streaming?

**Reply by USAMAWIZARD** [reply, marked as answer]
multitrack conference is a new implementation and the old conference implementation is deprecated.

**Reply by mekya** [reply]
Just let me add somethings,

`newStreamAvailable` is called for both audio and video tracks for compatibility reasons. 

It's good to listen `newTrackAvailable`.

Check this out -> https://github.com/ant-media/StreamApp/blob/master/src/main/webapp/js/webrtc_adaptor.js#L728

```javascript
             ...
            this.notifyEventListeners("newTrackAvailable", dataObj);

            //deprecated. Listen newTrackAvailable in callback. It's kept for backward compatibility
            this.notifyEventListeners("newStreamAvailable", dataObj);
           ....
```


FYI

**Reply by MBFLYT** [reply]
@USAMAWIZARD using multi track solved the issue.
